test/testcafe/spec/TerminalView_spec.js

```diff
@@ -195,44 +195,84 @@ test.requestHooks(terminalCustomAccessDeniedErrorMessageMock)('should render cus
   await t.expect(terminalViewPage.form.getErrorBoxHtml()).eql('<span data-se="icon" class="icon error-16"></span><div class="custom-access-denied-error-message"><p>You do not have permission to perform the requested action.</p><ul class="custom-links"><li><a href="https://www.okta.com/" target="_blank" rel="noopener noreferrer">Help link 1</a></li><li><a href="https://www.okta.com/help?page=1" target="_blank" rel="noopener noreferrer">Help link 2</a></li></ul></div>');
 });
 
+const assertAnchorWithBlankTargetHaveCorrectRelAttribute = async (t, selector) => {
+  await t.expect(
+    selector
+      .find('a[target="_blank"]')
+      .filter((node) => {
+        const relValues = (node.getAttribute('rel') || '').split(' ');
+        return relValues.indexOf('noopener') < 0 || relValues.indexOf('noreferrer') < 0;
+      })
+      .exists).eql(false);
```

Comment:
Trying to understand this logic.. let me know if this is correct:
1. You take the selector and find the anchor tag.
2. You filter through the nodes in the anchor tag until we find the rel attr
3. You check if either of the `noopener` or `noreferrer` tags index are -1. If -1, it implies that it's not present in the rel attr.
4. Assuming the html has the rel attr, this should always return an empty object, therefore exists should be false.

Ok I actually think I understand it, but just felt a bit convoluted since you're checking for the inverse. Maybe it would be clearer if you checked for `return relValues.indexOf('noopener') >= 0 || relValues.indexOf('noreferrer') >= 0;` and assert exists is true?

Not sure if there's any flakiness in checking for a hardcoded `0` and `1` index in the return statement, if no flakiness then that would be my suggestion.

Reply:
The goal of the test is to check:
- for all anchors that have `target="_blank"`
- filter for any that don't rel attributes with either value `noopener` or `noreferrer`
- if it exists, fail the the test.

The goal of the check is to assert not a single anchor with the missing attributes exist, as [it's considered bad](https://www.jitbit.com/alexblog/256-targetblank---the-most-underestimated-vulnerability-ever/)

Reply:
@vincentle-okta can you please move the selector/filter logic into the page object for the terminal view?

Reply:
Moved the selector to `getAnchorsWithBlankTargetsWithoutRelevantAttributes`.